danger-kotlin-library/src/main/kotlin/systems/danger/kotlin/GitLab.kt

```diff
@@ -1,93 +1,106 @@
+@file:UseSerializers(DateSerializer::class)
+
 package systems.danger.kotlin
 
-import com.squareup.moshi.Json
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.UseSerializers
+import systems.danger.kotlin.serializers.DateSerializer
 import java.util.*
 
+@Serializable
 data class GitLab(
-    @Json(name="mr")
+    @SerialName("mr")
     val mergeRequest: GitLabMergeRequest,
     val metadata: GitLabMetadata
 )
 
-data class GitLabDiffRefs (
-    @Json(name="base_sha")
+@Serializable
+data class GitLabDiffRefs(
+    @SerialName("base_sha")
     val baseSha: String,
-    @Json(name="head_sha")
+    @SerialName("head_sha")
     val headSha: String,
-    @Json(name="start_sha")
+    @SerialName("start_sha")
     val startSha: String
 )
 
+@Serializable
 data class GitLabUserMergeData(
-    @Json(name="can_merge")
+    @SerialName("can_merge")
     val canMerge: Boolean
 )
 
+@Serializable
 data class GitLabMergeRequest(
-    @Json(name="allow_collaboration")
+    @SerialName("allow_collaboration")
     val allowCollaboration: Boolean?,
-    @Json(name="allow_maintainer_to_push")
+    @SerialName("allow_maintainer_to_push")
     val allowMaintainerToPush: Boolean?,
```

Comment:
I suppose we could default those to false if not present and remove the optional